Lex/fee fetch loop

LPeterson18 opened the pull request:
1. Updated Databases to structure around partners
2. Added fetch calls for ethgasstation.info, mempool.space, and bitcoinfees.earn.com
3. Added engine to fetch from APIs and add collected fees to DBs with normalized dates as Ids

LPeterson18 commented:
Rename and restructure files and folders for better readability

LPeterson18 commented:
Re-organized file structure as detailed during PR review

thehobbit85 commented on src/feesEngine.ts:
```diff
@@ -0,0 +1,98 @@
+      } else if (feesResult[0] === null) {
+        mylog('Error getting eth fees from Eth Gas Station')
+      }
+      if (feesResult[0] != null) {
```
This should probably be `feesResult[1]` no?

thehobbit85 reviewed (changes requested):
See comment

LPeterson18 commented:
Fixed error handling

LPeterson18 commented:
Rebased branch

LPeterson18 commented on src/partners/earn.ts:
```diff
@@ -0,0 +1,58 @@
+      Accept: 'application/json'
+    }
+  }
+  const url = `https://bitcoinfees.earn.com/api/v1/fees/list`
```
Base URIs for all providers should be in config file

LPeterson18 commented on src/feesEngine.ts:
```diff
@@ -0,0 +1,98 @@
+  while (true) {
+    let preppedData
+    let prePrepData
+    const earn = earnCall()
+    const mempool = mempoolSpaceCall()
+    const ethGasStation = ethGasStationCall()
```
Handle result/rejected promises on each one and not on returned array from promise.all

LPeterson18 commented on src/feesEngine.ts:
```diff
@@ -0,0 +1,98 @@
+promisify(dbEarn)
+promisify(dbMempool)
+
+const snooze = async (ms: number): Promise<void> =>
+  await new Promise((resolve: Function) => setTimeout(resolve, ms))
```
Move to utils file

LPeterson18 commented:
1. Moved Base URIs to config
2. Changed engine to handle resolved/rejected promises on each one and not on returned array from promise.all
3. Moved snooze to utils
4. Reorganized commits

LPeterson18 commented:
1. Reduced fees processing logic into one function using cleaners

LPeterson18 commented:
Removed redundant code by creating generic fetch function

LPeterson18 commented:
Fixed eslint-disable-next-line and rebased branch
